### PR TITLE
WIP fix ignored omitempty in RerunAuthConfig

### PR DESCRIPTION
**Pull request by droslean**
after #13564 merged, `omitempty` is ignored in `RerunAuthConfig`. That is causes a job to be generated with empty `rerun_auth_config: {}`.

### CODE REVIEW

File: prow/apis/prowjobs/v1/types.go [hunk 1]
```diff
@@ -164,7 +164,7 @@ type ProwJobSpec struct {
 	ReporterConfig *ReporterConfig `json:"reporter_config,omitempty"`
 
 	// RerunAuthConfig holds information about which users can rerun the job
-	RerunAuthConfig RerunAuthConfig `json:"rerun_auth_config,omitempty"`
```
Comment (alvaroaleman) [inline prow/apis/prowjobs/v1/types.go]: The fix makes perfect sense, did you check that there are no other places where we access the `RerunAuthConfig` and would now need a nilcheck?

Also small nit: three commits for chaing 23 LOC is probably a bit too much

Comment (droslean) [reply prow/apis/prowjobs/v1/types.go]: thanks, checking...

**Comment by stevekuznetsov** [comment]
Where are we serializing ProwJobs? We very explicitly do not want this to be a pointer. The job config should be fine to generate the way it is

**Comment by droslean** [comment]
Closing, since we are not serializing prowjobs anywhere.
/close

**Pull request closed without merging**